### ISSUE TITLE
Thrown lua errors should look nice when caught by JS

### ISSUE THREAD

**Issue by daurnimator** [fengari-lua/fengari-interop#7]
If JS has a try/catch around an exposed lua function then it currently gets an ugly error object:

```lua
function js.global:lua_error() error("AN ERROR") end
js.global:eval("try { lua_error() } catch(e) { console.error(e) }")
```
```
Object { status: 2, previous: null }
```

**Comment by daurnimator** [comment]
Related? Via `Error.prepareStackTrace` we can make stack traces use lua frames nicely.
See https://github.com/v8/v8/wiki/Stack%20Trace%20API#customizing-stack-traces

**Comment by daurnimator** [comment]
Fixed via cd8cebaebd816c1aad7098d50333cfee165ac611
